src/main/java/se/liu/ida/hefquin/engine/queryproc/impl/optimizer/simple/RandomizedJoinPlanOptimizerImpl.java

```diff
@@ -0,0 +1,91 @@
+package se.liu.ida.hefquin.engine.queryproc.impl.optimizer.simple;
+
+import java.util.List;
+import java.util.Random;
+
+import se.liu.ida.hefquin.engine.queryplan.PhysicalPlan;
+import se.liu.ida.hefquin.engine.queryplan.utils.PhysicalPlanFactory;
+import se.liu.ida.hefquin.engine.queryproc.QueryOptimizationException;
+
+public class RandomizedJoinPlanOptimizerImpl extends JoinPlanOptimizerBase {
+	@Override // Implement initializeEnumerationAlgorithm from JoinPlanOptimizerBase.
+	public EnumerationAlgorithm initializeEnumerationAlgorithm( final List<PhysicalPlan> subplans ) {
+		return new RandomizedEnumerationAlgorithm(subplans);
+	}
+	
+	protected class RandomizedEnumerationAlgorithm implements EnumerationAlgorithm {
+		
+		/**
+		 * The class contains a random seed object.
+		 */
+		Random random;

```

Comment:
A few points here:

1. Member variables such as this one should typically be `protected`, which makes it possible to access them from within potential future subclasses.
2. The member variable should be `final`. This way, after it is initialized (e.g., in the constructor), we cannot assign something else to it later. The idea is that we do not intend to later override the value of this member variable with another `Random` object because this way. The keyword `final` allows us to make this intention explicit and to get a compiler error in case we do anything unexpected later. So, using `final` is mainly a safety measure and it is generally a good practice to use it.
3. For member variables that do not need anything from the constructor to be initialized, it is preferable to include their initialization already in the declaration rather than in the constructor.

So, based on all of these points, this line should be changed as follows:
```suggestion
		protected final Random random = new Random();
```
By the way, notice that you can accept these code change suggestions directly here in the PR, in which case a commit is created automatically. The only thing that you need to be mindful about if you do so is that, if you later also want to do more code changes in your IDE, you have to do a `git pull` first in order to update your local branch to include these commits.